Automattic/wp-calypso
# Show Google My Business tab only once connected

This pull request is a follow-up of https://github.com/Automattic/wp-calypso/pull/24015 that aims at:

* Enabling Google My Business back in the `development`, `wpcalypso`, and `stage` environments 
* Displaying the new stats tab only if a location has been connected
* Hiding the nudge on the `Stats` page if a location has been connected
* Grouping all Google My Business actions and selectors in the same place
* Removing some unused code

#### Testing instructions
 
1. Run `git checkout update/gmb` and start your server, or open a [live branch](https://calypso.live/?branch=update/gmb)
2. Log in to an account with a site eligible to Google My Business
3. Open the [`Stats` page](http://calypso.localhost:3000/stats)
4. Assert that no `Google My Business` tab is displayed in the header
5. Assert that the `Can your customers find you on Google?` nudge is displayed
6. Click the `Start Now` button on the nudge
7. Click the `Create Your Listing` button
8. Click the `Create Your Listing` button
9. Click the `Connect Location` button to access stats
10. Assert that a `Google My Business` tab is now displayed in the header
11. Click the `Traffic` tab
12. Assert that the nudge is no longer shown
13. Reload the page to restore initial state
14. Click the `X` icon in the upper right corner of the nudge
15. Assert that the nudge is no longer shown, even if you reload the page

#### Reviews
